src/net_processing.cpp

```diff
@@ -2040,12 +2040,9 @@ void static ProcessOrphanTx(CConnman& connman, CTxMemPool& mempool, std::set<uin
         const CTransactionRef porphanTx = orphan_it->second.tx;
         const CTransaction& orphanTx = *porphanTx;
         NodeId fromPeer = orphan_it->second.fromPeer;
-        // Use a new TxValidationState because orphans come from different peers (and we call
-        // MaybePunishNodeForTx based on the source peer from the orphan map, not based on the peer
-        // that relayed the previous transaction).
-        TxValidationState orphan_state;
+        TxValidationState state;
```

Comment:
in commit f98ffeb4323111f214234ef33:

I'd be surprised if this commit compiles. Especially L2094

Reply:
Good catch Marco. This was a bad rebase. I've fixed it and verified that all commits compile.